include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -90,7 +93,12 @@ def RegresetWithInvalidResetValue : Pat<
   []>;
 
 // regreset(clock, constant_zero, resetValue) -> reg(clock)
-def RegresetWithZeroReset : Pat<
+def RegresetWithZeroReset1 : Pat<
+  (RegResetOp $clock, (ConstantOp ZeroConstantAttr), $resetValue,
+   $name, $annotations),
+  (RegOp $clock, $name, $annotations)>;
```

Comment:
It seems like there should be some way to not have to duplicate the pattern by matching one of several types. However, I couldn't find any examples or documentation on how to do it.

Reply:
Good point. I think I could just combine two constraints, one for `SpecialConstantOp` and one for `ConstantOp`, into an op constraint. Let me give this a try.